safe_transaction_service/safe_messages/serializers.py

```diff
@@ -66,6 +66,11 @@ def validate(self, attrs):
             raise ValidationError(
                 f"1 owner signature was expected, {len(safe_signatures)} received"
             )
+        for safe_signature in safe_signatures:
+            if not safe_signature.is_valid(ethereum_client, safe_address):
+                raise ValidationError(
+                    f"Signature={safe_signature.signature.hex()} for owner={safe_signature.owner} is not valid"
+                )
```

Comment:
What is the reason for this check? Was it missing?

Maybe `SafeSignature.parse_signature` should throw in these cases or?

Reply:
> What is the reason for this check? Was it missing?
Yes

> Maybe SafeSignature.parse_signature should throw in these cases or?
No, as for EIP1271 and ApprovedHash signatures a network request is required to check, and maybe is not desired. So first we parse the signatures (only in structure) and then you can validate them

Reply:
> > Maybe SafeSignature.parse_signature should throw in these cases or?
> 
> No, as for EIP1271 and ApprovedHash signatures a network request is required to check, and maybe is not desired. So first we parse the signatures (only in structure) and then you can validate them

Ok in my opinion we should review these cases (if in the other cases they should throw a validation error or not). If the result of a "failed" parse can be ignored in some cases then I agree that we don't need to change it.

Reply:
But the problem is that parse can be valid but signature not, and that depends for example is a hash is approved on the contract, and you don't want to check the signatures when you parse them as blockchain calls are not really fast (we are using it in a lot of places as it's a core function from `safe-eth-py`)